AD_query_tool.py

```diff
@@ -78,22 +78,61 @@ def enabled_inactive_users():
+    df_enabled_accounts_without_password = df.loc[df['UserAccountControl'] & 0x0020 == 1] 
+    df_enabled_accounts_without_password_amount = df_enabled_accounts_without_password['CN'].count()
+    return(df_enabled_accounts_without_password_amount)
+
+def enabled_domain_admins():   
+    df = pd.DataFrame(df_including_password_age())
+    df_enabled_users = enabled_users(df)
+    search_list = ['Domain Admins', 'Administrator', 'Enterprise Admins']
+    df_domain_admins = df_enabled_users.loc[df_enabled_users['GroupsMembership'].str.contains('|'.join(search_list))]
+    df_domain_admins_amount = df_domain_admins['CN'].count()
+    return(df_domain_admins)
+
+def enabled_domain_admins_avg_password_age():
+    df_enabled_domain_admins = enabled_domain_admins()
+    avg_password_age = df_enabled_domain_admins['PasswordAge'].mean()
+    return(int(avg_password_age))
+
+def enabled_domain_admins_amount():   
+    df_domain_admins = enabled_domain_admins()
+    enabled_domain_admins_amount = df_domain_admins['CN'].count()
+    return(enabled_domain_admins_amount)
+
+# When SMARTCARD_REQUIRED flag is set, it forces the user to log on by using a smart card
+def enabled_admins_card_req():   
+    df_domain_admins = enabled_domain_admins()
+    admins_card_req = df_domain_admins.loc[df_domain_admins['UserAccountControl'] & 0x40000 == 1]
+    admins_card_req_amount = admins_card_req['CN'].count()
+    return(admins_card_req_amount)
+
+# When NOT_DELEGATED flag is set, the security context of the user isn't delegated to a service even if the service account is set as trusted for Kerberos delegation
+def enabled_admins_account_not_delegated():   
+    df_domain_admins = enabled_domain_admins()
+    admins_non_delegated = df_domain_admins.loc[df_domain_admins['UserAccountControl'] & 0x100000 == 1]
+    admins_non_delegated_amount = admins_non_delegated['CN'].count()
+    return(admins_non_delegated_amount)
 
 def main():
-    connection = active_directory_conn('ldap://172.20.1.216:389', 'Administrator', 'Nhwk2iISxJuOZQa43IDch0J8z0hqB5jY')   
-    active_directory_query = search_active_directory('DC=lab, DC=visiblerisk, DC=com', '(&(objectClass=user)(objectCategory=person))', ['userAccountControl', 'lastLogon', 'pwdLastSet', 'memberOf'], connection)
-    results_to_csv(csv_path, active_directory_query) 
-    print('\nAmount of enabled domain users :', enabled_users_amount(), '\n')
-    password_age() 
-    print('\nAvarege password age of enabled domain admins:',enabled_domain_admins_avg_password_age(), 'days\n')
+    connection = active_directory_conn(LDAP_SERVER, LDAP_LOGIN, LDAP_PASS)   
+    active_directory_output = search_active_directory(BASE_DN, SEARCH_OBJECT, SEARCH_ATTR, connection)
+    results_to_csv(CSV_PATH, active_directory_output) 
+    print('\nAmount of enabled domain admins:', enabled_domain_admins_amount())
+    print('\nAvarege password age of enabled domain admins:',enabled_domain_admins_avg_password_age(), 'days')
```

Comment:
just for fun, try to use this trick for string formatting: https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python